coorx/image.py

```diff
@@ -44,117 +44,126 @@
 
 from .coordinates import Point, PointArray
 from .linear import AffineTransform
-from .systems import CoordinateSystemGraph
+from .systems import CoordinateSystemGraph, CoordinateSystem
 
 
 class Image:
-    """Wraps image data with a coordinate system and methods 
+    """Wraps image data with a coordinate system and methods
     for transforming the image and mapping coordinates through the transforms.
 
     Parameters
     ----------
     image : ndarray
         The image data. Must be 2D or higher.
-    axes : tuple | None
-        The axis indices of the row and column dimensions in the image data.
-        If *image* is 2D, then *axes* defaults to (0, 1).
-    cs_name : str | None
+    axes : tuple, optional
+        The axes of the image that correspond to spatial dimensions. Defaults to all axes.
+    cs_name : str | CoordinateSystem | None
         Optional name of the coordinate system to attach to the image.
     graph : str | CoordinateSystemGraph | None
         Optional graph to use for the coordinate system.
     """
-    _image_graph_n = 0
 
-    def __init__(self, image, axes: tuple | None = None, cs_name=None, graph=None):
-        self.image = image
-        self._parent_tr = None
+    def __init__(self, image, axes=None, cs_name=None, graph=None):
         if axes is None:
-            if image.ndim == 2:
-                axes = (0, 1)
-            else:
-                raise ValueError("Must specify (row, col) axes for image data with more than 2 dimensions")
+            axes = tuple(range(image.ndim))
         self.axes = axes
 
+        self.image = image
+        self._parent_tr = None
+
         if graph is None:
             graph = 'image_graph'
-            Image._image_graph_n += 1
```

Comment:
The intent here (although not fully implemented) was to have a different system for each image by default.. I can see that being useful or not depending on the case, but maybe leaning towards that?

Reply:
ticketed in #43